lib/charms/opensearch/v0/opensearch_profile.py

```diff
@@ -234,34 +208,55 @@ def check_cluster_topology(self, profile: OpenSearchProfile) -> List[str]:
             app.planned_units for app in cluster_fleet_apps.values() if "data" in app.roles
         )
 
-        if nbr_cm_nodes < profile.cluster_topology_requirements.cluster_managers:
-            missing_requirements.append(
-                f"At least {profile.cluster_topology_requirements.cluster_managers} cluster manager nodes are required. Found only {nbr_cm_nodes}."
-            )
-
-        if nbr_data_nodes < profile.cluster_topology_requirements.data:
-            missing_requirements.append(
-                f"At least {profile.cluster_topology_requirements.data} data nodes are required. Found only {nbr_data_nodes}."
-            )
-
-        return missing_requirements
-
-    def check_all_requirements(self, profile: Optional[OpenSearchProfile] = None) -> List[str]:
+        match nbr_cm_nodes < profile.cluster_topology_requirements.cluster_managers, nbr_data_nodes < profile.cluster_topology_requirements.data:
+            case (True, True):
+                error_message = f"At least {profile.cluster_topology_requirements.cluster_managers} cluster manager nodes and {profile.cluster_topology_requirements.data} data nodes are required."
+            case (True, False):
+                error_message = f"At least {profile.cluster_topology_requirements.cluster_managers} cluster manager nodes are required."
+            case (False, True):
+                error_message = f"At least {profile.cluster_topology_requirements.data} data nodes are required."
+            case _:
+                return []
+
+        logger.error("Missing cluster topology requirements: %s", error_message)
+        return [error_message]
+
+    def check_all_requirements(
```

Comment:
We should rename this method, as we are checking for missing requirements (as also returned)